src/Neo/SmartContract/KeyBuilder.cs

```diff
@@ -79,9 +84,10 @@ public KeyBuilder Add(ISerializable key)
         /// </summary>
         /// <param name="key">Part of the key.</param>
         /// <returns>A reference to this instance after the add operation has completed.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public KeyBuilder AddBigEndian(int key)
         {
-            var data = new byte[sizeof(int)];
+            Span<byte> data = stackalloc byte[sizeof(int)];
```

Comment:
Are you sure this makes any difference? Have you measured these changes in isolation?

Reply:
> Are you sure this makes any difference? Have you measured these changes in isolation?

Yes. This change avoids allocating memory from the heap.

Reply:
> In general compilers can optimize such allocations out (but I'm not aware of C# compiler specifics, maybe it never does it and I just don't know), but if you've checked that this particular change is really worthwhile (and this means testing it in isolation from other changes in the same PR) then OK.

I not found that C# no such allocation optimization(Golang can optimize it in some cases). And I added a new benchmark:
```
BenchmarkDotNet v0.13.12, macOS 15.1.1 (24B2091) [Darwin 24.1.0]
Apple M4 Pro, 1 CPU, 14 logical and 14 physical cores
.NET SDK 8.0.404
  [Host]     : .NET 8.0.11 (8.0.1124.51707), Arm64 RyuJIT AdvSIMD
  DefaultJob : .NET 8.0.11 (8.0.1124.51707), Arm64 RyuJIT AdvSIMD


| Method                           | Mean     | Error    | StdDev   |
|--------------------------------- |---------:|---------:|---------:|
| KeyBuilder_AddInt                | 26.30 ns | 0.116 ns | 0.109 ns |
| KeyBuilder_AddIntWithoutPrealloc | 32.40 ns | 0.084 ns | 0.079 ns |
| KeyBuilder_AddBytes              | 26.84 ns | 0.043 ns | 0.034 ns |
| KeyBuilder_AddUInt160            | 25.11 ns | 0.034 ns | 0.030 ns |
```